frontend/packages/metal3-plugin/src/components/baremetal-hosts/add-baremetal-host/AddBareMetalHost.tsx

```diff
@@ -77,13 +109,17 @@ const AddBareMetalHost: React.FC<AddBareMetalHostProps> = ({ namespace, hosts })
   };
 
   return (
-    <Formik
-      initialValues={initialValues}
-      onSubmit={handleSubmit}
-      onReset={history.goBack}
-      validationSchema={addHostValidationSchema}
-      component={AddBareMetalHostForm}
-    />
+    <AddBareMetalHostContext.Provider value={{ isNameDisabled: isEditing, isEditing, showUpdated }}>
+      <Formik
+        initialValues={initialValues}
+        enableReinitialize={isEditing && (reload || !prevInitialHost || !prevInitialSecret)}
+        onSubmit={handleSubmit}
+        onReset={() => setReload(true)}
+        validationSchema={addHostValidationSchema}
+        component={AddBareMetalHostForm}
+        isNameDisabled={isEditing}
+      />
```

Comment:
The context is not needed. By using `children` instead of `component` prop, it is possible to pass additional props.
```suggestion
      >
        {(props) => (
          <AddBareMetalHostForm {...props} showUpdated={showUpdated} isEditing={isEditing} />
        )}
      </Formik>
```